### PR TITLE
Add button to call 511 directly for transit predictions

### PR DESCRIPTION
**Pull request by mattestrada01**
Users are now able to use a dynamic call function that gives them updated time information for designated stops. This works for all stations except BART.

### CODE REVIEW

File: frontend/src/TransitPredictionsPage.js [hunk 1]
```diff
@@ -112,29 +112,50 @@ export default function TransitPredictionsPage() {
...
-              <div className="flex justify-center xl:justify-start font-bold text-[1rem] md:text-4xl mb-5 xl:mb-10">{stop.name}</div>
+            <div key={stop.name} className="items-center text-center md:text-left flex flex-col w-full xl:w-full mt-2 p-6 min-w-50 max-h-[65vh] text-xl bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700 overflow-y-auto">
+              
+              <div className="flex justify-center xl:justify-start font-bold text-[1rem] md:text-4xl mb-2 xl:mb-5">
+                {stop.name}
+              </div>
+              <div className="flex justify-center xl:justify-start text-sm md:text-lg text-gray-500 mb-4 xl:mb-6">
+                {stop.ids.length > 0 ? (
+                  <>
+                    {stop.ids.some(id => !isNaN(id)) ? (
```
Comment (evanugarte) [inline frontend/src/TransitPredictionsPage.js]: the logic for finding which stop id is a number should live where we were setting the id fields above, on this line

```
ids: Array.isArray(stop.ids) ? stop.ids : [],
```

we can make a field like `phoneCallStopId` to be the first numerical stop id

File: frontend/src/TransitPredictionsPage.js [hunk 2]
```diff
@@ -112,29 +112,50 @@ export default function TransitPredictionsPage() {
...
+                      <a
+                        href={`tel:511p1p1,,${stop.ids.find(id => !isNaN(id))}`}
+                        className="inline-block px-2 py-1 ml-2 text-white bg-blue-600 border border-blue-600 rounded hover:bg-blue-700"
+                      >
+                        CLICK TO CALL
+                      </a>
+                    ) : (
+                      <span className="inline-block px-2 py-1 ml-2 text-gray-500 border border-gray-500 rounded">NO CALL AVAILABLE</span>
```
Comment (evanugarte) [inline frontend/src/TransitPredictionsPage.js]: instead lets just not show anything if the call isnt available

File: frontend/src/TransitPredictionsPage.js [hunk 3]
```diff
@@ -112,29 +112,50 @@ export default function TransitPredictionsPage() {
...
+                    )}
+                  </>
+                ) : (
+                  <span>No IDs</span>
+                )}
```
Comment (evanugarte) [inline frontend/src/TransitPredictionsPage.js]: do we need this

File: frontend/src/TransitPredictionsPage.js (hunk 2 above)
Comment (mattestrada01) [reply frontend/src/TransitPredictionsPage.js]: fixed

File: frontend/src/TransitPredictionsPage.js (hunk 3 above)
Comment (mattestrada01) [reply frontend/src/TransitPredictionsPage.js]: removed

File: frontend/src/TransitPredictionsPage.js (hunk 1 above)
Comment (mattestrada01) [reply frontend/src/TransitPredictionsPage.js]: resolved with new method and updated arrays for app.py

**Pull request merged**